rust/read_file.rs

```diff
@@ -0,0 +1,18 @@
+use std::env;
+use std::fs::File;
+use std::io::{BufReader, BufRead};
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    let file_path = &args[1];
+
+    let file = File::open(file_path).unwrap();
+    let reader = BufReader::new(file);
+
+    for (mut line_number, line_result) in reader.lines().enumerate() {
+        line_number += 1;
+        let line = line_result.unwrap();
+        println!("{} {}", line_number, line.to_uppercase());
```

Comment:
```suggestion
    for (line_number, line_result) in reader.lines().enumerate() {
        let line = line_result.unwrap();
        println!("{} {}", line_number + 1, line.to_uppercase());
```

Generally, we should prefer non-mutable variables.